File: bank_statement/citi.py
import sys


def convert_format(file_path):
    with open(file_path, 'r') as file:
        input_text = file.read()
    output_text = []
    lines = input_text.split('\n')
    for line in lines:
        components = line.split()
        date = components[0]
        desc = ' '.join(components[1:-1])
        amount = components[-1]
        if amount.startswith('(') and amount.endswith(')'):
            amount = '-' + amount[1:-1]
        output_text.append(f"{date};{desc};{amount}")
    return '\n'.join(output_text)


if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python convert_format.py <file_path>")
    else:
        file_path = sys.argv[1]
        print(convert_format(file_path))

File: credit/citi.go
package credit

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func ConvertCiti(filePath string) (string, error) {
	file, err := os.Open(filePath)
	if err != nil {
		return "", err
	}
	defer file.Close()

	var outputText []string
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		components := strings.Fields(line)
		date := components[0]
		desc := strings.Join(components[1:len(components)-1], " ")
		amount := components[len(components)-1]
		if strings.HasPrefix(amount, "(") && strings.HasSuffix(amount, ")") {
			amount = "-" + strings.Trim(amount, "()")
		}
		outputText = append(outputText, fmt.Sprintf("%s;%s;%s", date, desc, amount))
	}

	if err := scanner.Err(); err != nil {
		return "", err
	}

	return strings.Join(outputText, "\n"), nil
}
